### ISSUE TITLE
HA frontend doesn't show devices, despite z2m is up'nd running

### ISSUE THREAD

**Issue by rainbowbreeze** [Koenkk/zigbee2mqtt#12575]
### What happened?

I'm connected to my LAN, and I'm using HA Supervisor. z2m addon correctly installed and running, and everything worked 1-2 weeks ago. Today, If I try to open z2m frontend from within HA, it doesn't fully work, with different scenarios. 

- If I use the FQDN within my LAN, the z2m frontend is visible, but no devices are shown, all the input fields in other many entries are empty, etc. So, apart from the "header" of z2m frontend, nothing else is shown in my browser window.
![Screen Shot 2022-05-23 at 15 37 27](https://user-images.githubusercontent.com/381940/169832875-afdcd8c1-2e24-4b63-9f8a-59a0f3356650.png)
- If I use IP adress to open HA (192.168.xx.yy), not even the frontend is shown. Complete blank screen (or white, depends on our theme)

### Chrome web tools output
- When I access HA via FQDN
![Screen Shot 2022-05-23 at 15 59 37](https://user-images.githubusercontent.com/381940/169836855-f26081c7-f6bc-43d4-84a9-be4876193ea0.png)
- When I access HA via FQDN
`192.168.100.XXX/:1  GET http://192.168.100.XXX:8123/api/hassio_ingress/7u_fnof4bjH7vzK664s_9Ct2SJ-qNSL0Wtgn7D0PJk0/scripts/vendor.f3635c6dc09aacfa6bae.js net::ERR_ABORTED 404 (Not Found)
`

### Config
z2m configuration.yaml
```
homeassistant: true
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://core-mosquitto:1883
  user: addons
  password: -------obscured-----
serial:
  port: /dev/ttyACM0
  adapter: deconz
frontend: true
advanced:
  log_level: debug
availability:
  active:
    timeout: 10
  passive:
    timeout: 720
devices:
  '0x588e81fffeff5e88':
[...]
```

HA z2m add on
```
data_path: /config/zigbee2mqtt
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: '-d -d'
  log: false
mqtt: {}
serial: {}
```

### Logs
z2m
```
debug 2022-05-23 15:07:20: Loaded state from file /config/zigbee2mqtt/state.json
info  2022-05-23 15:07:20: Logging to console and directory: '/config/zigbee2mqtt/log/2022-05-23.15-07-17' filename: log.txt
debug 2022-05-23 15:07:20: Removing old log directory '/config/zigbee2mqtt/log/2022-04-26.23-07-26'
info  2022-05-23 15:07:20: Starting Zigbee2MQTT version 1.25.1 (commit #unknown)
info  2022-05-23 15:07:20: Starting zigbee-herdsman (0.14.27)
debug 2022-05-23 15:07:20: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/config/zigbee2mqtt/coordinator_backup.json","databaseBackupPath":"/config/zigbee2mqtt/database.db.backup","databasePath":"/config/zigbee2mqtt/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":6754},"serialPort":{"adapter":"deconz","path":"/dev/ttyACM0"}}'
info  2022-05-23 15:07:20: zigbee-herdsman started (resumed)
info  2022-05-23 15:07:20: Coordinator firmware version: '{"meta":{"maintrel":0,"majorrel":38,"minorrel":114,"product":0,"revision":"0x26720700","transportrev":0},"type":"ConBee2/RaspBee2"}'
debug 2022-05-23 15:07:20: Zigbee network parameters: {"channel":11,"extendedPanID":"0xdddddddddddddddd","panID":6754}
info  2022-05-23 15:07:20: Currently 15 devices are joined:
info  2022-05-23 15:07:20: socket_003_bw-shp13 (0x842e14fffe366708): TS0121_plug - TuYa 10A UK or 16A EU smart plug (Router)
[...]
info  2022-05-23 15:07:20: 0x0004740000939ee8 (0x0004740000939ee8): 067775/741811 - Legrand Power socket with power consumption monitoring (Router)
info  2022-05-23 15:07:20: Zigbee: disabling joining new devices.
info  2022-05-23 15:07:20: Connecting to MQTT server at mqtt://core-mosquitto:1883
debug 2022-05-23 15:07:20: Using MQTT login with username: addons
info  2022-05-23 15:07:21: Connected to MQTT server
info  2022-05-23 15:07:21: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'online'
debug 2022-05-23 15:07:21: Received MQTT message on 'zigbee2mqtt/bridge/info' with data
[...]
debug 2022-05-23 15:07:21: Received MQTT message on 'zigbee2mqtt/bridge/devices' with data '[{"definition":null,"endpoints":{"1":{"bindings":[],"clusters":{"input":["genBasic","genTime","genOta"],"output":["genPowerCfg","genPollCtrl","ssIasZone"]},"configured_reportings":[],"scenes":[]},"242":{"bindings":[],"clusters":{"input":[],"output":["greenPower"]},"configured_reportings":[],"scenes":[]}},"friendly_name":"Coordinator","ieee_address":"0x00212effff05fa4c","interview_completed":true,"interviewing":false,"network_address":0,"supported":false,"type":"Coordinator"},{"date_code":"","definition":{"description":"10A UK or 16A EU smart plug","exposes":[{"features":[{"access":7,"description":"On/off state of the
[...]
debug 2022-05-23 15:07:21: Received MQTT message on 'zigbee2mqtt/bridge/groups' with data '[{"friendly_name":"default_bind_group","id":901,"members":[],"scenes":[]}]'
debug 2022-05-23 15:07:21: Received MQTT message on 'zigbee2mqtt/bridge/extensions' with data '[]'
debug 2022-05-23 15:07:21: Received MQTT message on 'zigbee2mqtt/bridge/config' with data '{"commit":"unknown","coordinator":{"meta":{"maintrel":0,"majorrel":38,"minorrel":114,"product":0,"revision":"0x26720700","transportrev":0},"type":"ConBee2/RaspBee2"},"log_level":"debug","network":{"channel":11,"extendedPanID":"0xdddddddddddddddd","panID":6754},"permit_join":false,"version":"1.25.1"}'
debug 2022-05-23 15:07:21: Received MQTT message on 'zigbee2mqtt/901/availability' with data 'online'
debug 2022-05-23 15:07:21: Received MQTT message on 'zigbee2mqtt/0x0004740000937ec5/availability' with data 'online'
[...]
info  2022-05-23 15:07:22: MQTT publish: topic 'zigbee2mqtt/901/availability', payload 'online'
info  2022-05-23 15:07:22: Started frontend on port 0.0.0.0:8099
debug 2022-05-23 15:07:22: Received MQTT message on 'zigbee2mqtt/light_02_ikea-e27-806lm/availability' with data 'offline'
```

HA Supervisor logs
```
22-05-23 15:28:12 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from core_ssh
22-05-23 15:28:12 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from core_ssh
22-05-23 15:28:12 INFO (MainThread) [supervisor.api.middleware.security] /host/info access from core_ssh
22-05-23 15:28:12 INFO (MainThread) [supervisor.api.middleware.security] /core/info access from core_ssh
22-05-23 15:30:45 INFO (MainThread) [supervisor.ingress] Update Ingress as panel for 45df7312_zigbee2mqtt
22-05-23 15:30:53 INFO (MainThread) [supervisor.ingress] Update Ingress as panel for 45df7312_zigbee2mqtt
22-05-23 15:34:13 ERROR (MainThread) [supervisor.api.ingress] Ingress error: [Errno 104] Connection reset by peer
22-05-23 15:34:35 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Server disconnected
22-05-23 15:37:45 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING
22-05-23 15:37:45 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
22-05-23 15:37:45 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON
22-05-23 15:37:45 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM
22-05-23 15:37:45 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.DNS_SERVER_IPV6_ERROR/ContextType.DNS_SERVER

```

### Zigbee2MQTT version

 1.25.1-1

### Adapter firmware version

0x26720700

### Adapter

Conbee II

**Comment by rainbowbreeze** [comment]
After some digging, I see it's a common problem:
- https://www.reddit.com/r/homeassistant/comments/ul4gej/zigbee2mqtt_web_ui_not_loading/
- https://www.reddit.com/r/homeassistant/comments/ukvigo/zigbee2mqtt_web_ui_does_not_show_up_anymore
- and others...

Overall, cleanig the browser cache / android app cache allows to access the z2m full frontend again, with device list, configs, etc.
I also tried to access HA in a new incognito browser window (both Chrome and Firefox on my mac machine), and I can access z2m flawlessly.

At this point, I suspect there is some sort of bug in cache management somewhere...

**Comment by ruant** [comment]
This happens to me sporadically too. Very annoying.
Just did a complete clean reinstall of HA and z2m addon and still issues.

**Comment by gitrummy** [comment]
I am also having this exact same issue. In my supervisor logs...

`2024-06-06 18:44:13.622 ERROR (MainThread) [supervisor.api.ingress] Ingress error: 404, message='Invalid response status', url=URL('http://172.30.33.5:8099/api')
2024-06-06 18:44:18.624 ERROR (MainThread) [supervisor.api.ingress] Ingress error: 404, message='Invalid response status', url=URL('http://172.30.33.5:8099/api')
2024-06-06 18:44:24.625 ERROR (MainThread) [supervisor.api.ingress] Ingress error: 404, message='Invalid response status', url=URL('http://172.30.33.5:8099/api')
2024-06-06 18:44:29.629 ERROR (MainThread) [supervisor.api.ingress] Ingress error: Server disconnected`

**Comment by gitrummy** [comment]
I resolved this issue, for me it was my custom header config on my Synology reverse proxy to HA. I had configured the custom headers manually some way I saw in the home assistant forums. I deleted the two lines I had and then did create -> WebSocket. It created two new lines that looked similar to the old ones but with slightly different wording, and then ingress of zigbee2mqtt worked instantly. 
![image](https://github.com/Koenkk/zigbee2mqtt/assets/11849147/f8fa83b8-ea3c-4233-99b2-f68e70883b11)
